nvflare/fuel/f3/streaming/byte_receiver.py

```diff
@@ -140,6 +145,26 @@ def register_callback(self, channel: str, topic: str, stream_cb: Callable, *args
 
         self.registry.set(channel, topic, Callback(stream_cb, args, kwargs))
 
+    def stop_task(self, task: RxTask, error: StreamError = None, notify=True):
+        if error:
+            log.error(f"Stream error: {error}")
+            task.stream_future.set_exception(error)
+
+            if notify:
+                message = Message()
+
+                message.add_headers(
+                    {
+                        StreamHeaderKey.STREAM_ID: task.sid,
+                        StreamHeaderKey.DATA_TYPE: StreamDataType.ERROR,
+                        StreamHeaderKey.ERROR_MSG: str(error),
+                    }
+                )
+                self.cell.fire_and_forget(STREAM_CHANNEL, STREAM_ACK_TOPIC, task.origin, message)
+
+        task.eos = True
+        self.rx_task_map.pop(task.sid)
```

Comment:
Same as other one, need to catch exception for the potential exception in self.cell.fire_and_forget().

Reply:
Agreed. The whole thing should be placed in a "try/except" block and place the "pop" in "finally". Basically we need to guarantee that the task is removed from the map no matter what happens.

Reply:
pop is moved to top and protected by a lock.